heartbeat/monitors/factory.go

```diff
@@ -140,3 +148,40 @@ func newCommonPublishConfigs(info beat.Info, cfg *common.Config) (pipetool.Confi
 		return clientCfg, nil
 	}, nil
 }
+
+func setupIndexProcessor(info beat.Info, settings publishSettings) (processors.Processor, error) {
+	var indexProcessor processors.Processor
+	if settings.DataStream != nil {
+		namespace := settings.DataStream.Namespace
+		if namespace == "" {
+			namespace = "default"
+		}
+		typ := settings.DataStream.Type
+		if typ == "" {
+			typ = "generic"
```

Comment:
Shouldn't this be `synthetics`, such that the index name becomes `synthetics-...` by default?

Reply:
Yes, I think I misunderstood a previous discussion of ours, glad to change it